src/services/parser.service.ts

```diff
@@ -0,0 +1,14 @@
+import { parse, prepareParserConfig } from '../utils/parser';
+
+/**
+ * Service providing `@asyncapi/parser` functionality.
+ */
+export class ParserService {
+  public async parse(asyncapi: string | any, options?: ReturnType<typeof prepareParserConfig>) {
+    if (!asyncapi) {
```

Comment:
This is also checked by `parse` Parser-JS method. Not sure if we want to leave the validation responsibility to the parser so we are consistent or what. See https://github.com/asyncapi/parser-js/blob/fb0ba2ebce8739371de179d825c357b6164fe0af/lib/utils.js#L100-L106

Reply:
okay, could we just remove the `ParserService`? It doesn't have any additional logic other than calling the `parse` function 😅 
I ended up not putting the try catch on this function because the code calling this service might want to handle errors differently. If this was to be reused in the CLI, a `ProblemException` and HTTP status codes might not make sense in that context. Therefore I think having this abstraction (ParserService) isn't adding value, just indirection.
@smoya and @magicmatatjahu what do you think?

Reply:
Yeah, if it's unnecessary, feel free to remove it :)